src/index.ts

```diff
@@ -1,174 +1,41 @@
-import { serialize, parse } from 'cookie';
-import type {
-  OptionsType,
-  TmpCookiesObj,
-  CookieValueTypes,
-  AppRouterCookies,
-  DefaultOptions,
-  CookiesFn,
-} from './types';
-import type { NextRequest, NextResponse } from 'next/server';
-export { CookieValueTypes } from './types';
-
-const isClientSide = (): boolean => typeof window !== 'undefined';
-
-const isCookiesFromAppRouter = (
-  cookieStore: TmpCookiesObj | AppRouterCookies | undefined,
-): cookieStore is AppRouterCookies => {
-  if (!cookieStore) return false;
-  return (
-    'getAll' in cookieStore &&
-    'set' in cookieStore &&
-    typeof cookieStore.getAll === 'function' &&
-    typeof cookieStore.set === 'function'
-  );
-};
-
-const isContextFromAppRouter = (
-  context?: OptionsType,
-): context is { res?: NextResponse; req?: NextRequest; cookies?: CookiesFn } => {
-  return (
-    (!!context?.req && 'cookies' in context.req && isCookiesFromAppRouter(context?.req.cookies)) ||
-    (!!context?.res && 'cookies' in context.res && isCookiesFromAppRouter(context?.res.cookies)) ||
-    (!!context?.cookies && isCookiesFromAppRouter(context.cookies()))
-  );
-};
-
-const transformAppRouterCookies = (cookies: AppRouterCookies): TmpCookiesObj => {
-  let _cookies: Partial<TmpCookiesObj> = {};
-
-  cookies.getAll().forEach(({ name, value }) => {
-    _cookies[name] = value;
-  });
-  return _cookies;
-};
-
-const stringify = (value: any) => {
-  try {
-    if (typeof value === 'string') {
-      return value;
+import * as clientCookies from './client';
+import * as serverCookies from './server';
+export * from './common/types';
+import type { OptionsType } from './common/types';
+import { isClientSide } from './common/utils';
+
+const getRenderPhase = () => (typeof window === 'undefined' ? 'server' : 'client');
+
+const getFn = (fnName: keyof typeof clientCookies, options?: OptionsType) => {
+  if (isClientSide(options)) {
+    if (getRenderPhase() === 'server') {
```

Comment:
```ts
import { getCookie } from 'cookies-next'

export const TestComponent = () => {
    const isActive = getCookie('active');
    
    return (
        <div>{isActive ? 'active': 'not active'}</div>
    );
}
```
it looks like we will have a difference between what the server and the client render. 

Because getCookie always returns undefined on the server phase.

Reply:
Oh, although the current version most likely has the same behavior

Reply:
In my implementation, we simply return `undefined` when a function is designated for client-side use, and the client component calling it is being pre-rendered.

Reply:
It's worth mentioning that `hasCookie` always returns a `boolean`. In a client component during the server phase, it returns `false` instead of `undefined.` In my opinion, having a consistent return type is important for `hasCookie`.

Reply:
I see the issue now. One part of the problem is the mismatch between server and client. Additionally, if we use a function from the `cookies-next` import, we also encounter type-related issues:

Do you have any ideas on how to fix this? 

![image](https://github.com/user-attachments/assets/3417adcd-eafb-4a64-8a19-1a20e5e60a8d)